medicines/doc-index-updater/src/storage_client/mod.rs

```diff
@@ -1,15 +1,54 @@
-use azure_sdk_core::errors::AzureError;
-use azure_sdk_storage_core::prelude::Client;
+use async_trait::async_trait;
+use azure_sdk_core::{errors::AzureError, prelude::*, DeleteSnapshotsMethod};
+use azure_sdk_storage_blob::Blob;
+use azure_sdk_storage_core::prelude::*;
 
-pub fn factory() -> Result<Client, AzureError> {
+pub struct BlobClient {
+    pub azure_client: Client,
+}
+
+impl BlobClient {
+    pub fn new(azure_client: Client) -> BlobClient {
+        BlobClient { azure_client }
+    }
+}
+
+#[async_trait]
+pub trait DeleteBlob {
+    async fn delete_blob(
+        &mut self,
+        container_name: &str,
+        blob_name: &str,
+    ) -> Result<(), AzureError>;
+}
+
+#[async_trait]
+impl DeleteBlob for BlobClient {
+    async fn delete_blob(
+        &mut self,
+        container_name: &str,
+        blob_name: &str,
+    ) -> Result<(), AzureError> {
+        self.azure_client
+            .delete_blob()
+            .with_container_name(&container_name)
+            .with_blob_name(&blob_name)
+            .with_delete_snapshots_method(DeleteSnapshotsMethod::Include)
+            .finalize()
+            .await?;
+        Ok(())
```

Comment:
is there no value in returning the `DeleteBlobResponse`?

Reply:
Not currently.